src/plugins/importFromJSONAction/ImportFromJSONAction.js

```diff
@@ -0,0 +1,253 @@
+
+            const compositionCollection = this.openmct.composition.get(domainObject);
+            let domainObjectKeyString = this.openmct.objects.makeKeyString(domainObject.identifier);
+            this.openmct.objects.mutate(rootObj, 'location', domainObjectKeyString);
+            compositionCollection.add(rootObj);
+        } else {
+            const dialog = this.openmct.overlays.dialog({
+                iconClass: 'alert',
+                message: "We're sorry, but you cannot import that object type into this object.",
+                buttons: [
+                    {
+                        label: "Ok",
+                        emphasis: true,
+                        callback: function () {
+                            dialog.dismiss();
+                        }
+                    }
+                ]
+            });
+        }
+    }
+    /**
+     * @private
+     * @param {object} rootModel
+     * @returns {object}
+     */
+    async _instantiate(rootModel) {
+        const success = await this.openmct.objects.save(rootModel);
+        if (success) {
+            return rootModel;
+        }
+
+        this.openmct.notifications.error('Error saving objects');
+    }
+    /**
+     * @private
+     * @param {object} oldId
+     * @param {object} newId
+     * @param {object} tree
+     * @returns {object}
+     */
+    _rewriteId(oldId, newId, tree) {
+        let newIdKeyString = this.openmct.objects.makeKeyString(newId);
+        let oldIdKeyString = this.openmct.objects.makeKeyString(oldId);
+        tree = JSON.stringify(tree).replace(new RegExp(oldIdKeyString, 'g'), newIdKeyString);
+
+        return JSON.parse(tree, (key, value) => {
+            if (value !== undefined
+                && value !== null
+                && Object.prototype.hasOwnProperty.call(value, 'key')
+                && Object.prototype.hasOwnProperty.call(value, 'namespace')
+                && value.key === oldId.key
+                && value.namespace === oldId.namespace) {
+                return newId;
+            } else {
+                return value;
+            }
+        });
+    }
+    /**
+     * @private
+     * @param {object} domainObject
+     * @param {object} parentDomainObject
+     */
+    _showForm(domainObject, parentDomainObject) {
+        const formStructure = {
+            title: this.name,
+            sections: [
+                {
+                    rows: [
+                        {
+                            name: 'Select File',
+                            key: 'selectFile',
+                            control: 'file-input',
+                            required: true,
+                            text: 'Select File...',
+                            validate: this._validateJSON,
+                            type: 'application/json'
+                        }
+                    ]
+                }
+            ]
+        };
+
+        this.openmct.forms.showForm(formStructure, {
```

Comment:
@nikhilmandlik Remind me why all forms need a `domainObject` and a `parentDomainObject`? In fact, in this case it doesn't seem as though any domain objects should be needed?